Source/Continent/Continent.cpp

```diff
@@ -22,36 +23,68 @@ Continent::Continent(
 {
 	this->size = size;
 	this->name = name;
-	initializeShape();
+	this->randomGenerator = randomGenerator;
+
+	InitializeShape(
+			humanCount,
+			zombieCount,
+			trapCount,
+			resourceCount);
 }
 
 Continent::~Continent()
 {
-	for(int i = 0; i < size; i++) {
-		for(int j = 0; j < size; j++) {
-			delete shape[i][j];
+	for(int y = 0; y < size; y++) {
+		for(int x = 0; x < size; x++) {
+			delete shape[y][x];
 		}
-		delete[] shape[i];
+		delete[] shape[y];
 	}
 	delete[] shape;
 }
 
 void Continent::Tick()
 {
+	Cell **list = new Cell*[size*size];
+
+	int count = 0;
+	for(int y = 0; y < size; y++) {
+		for(int x = 0; x < size; x++) {
+			Cell *current = shape[y][x];
+
+			list[count] = current;
+			count++;
+		}
+	}
 
+	for(int i = 0; i < count; i++)
+	{
+		Cell *current = list[i];
+		current->Tick();
+
+		CheckMove(current);
```

Comment:
In the CheckMove function we will need to add some code so that when a zombie moves into a trap the trap defense is reduced. Also are gonna implement this part assuming that when the zombie finds a trap it would die, or would its defense be reduced instead. Maybe we'll need to discuss this more

Reply:
@arive298 Yes, we have to discuss those final details. I implemented it this way, so that it is simpler. To me it makes more sense that the zombie is destroyed when it hits the trap.